README.md

```diff
@@ -1,16 +1,241 @@
-# Title
+# Nested Structures
 
 ## Learning Goals
 
--SWBAT 1
--SWBAT 2
+- Identify nested structures can be mixed
 
 ## Introduction
 
-## SWBAT 1
+You've spent the last few lessons learning about three basic nested structures:
 
-## SWBAT 2
+* `Array` of `Array`s, or "AoA"
+* `Array` of `Hash`es, or "AoH"
+* `Hash` of `Hash`es, or "HoH"
+* `Hash` of `Array`s, or "HoA"
+
+By using these basic nested data structures, we've gained a foundation for
+modeling many things in the world: vending machines, fictional characters,
+the genealogy of a Hollywood family, and a medical record result-set.
+
+But brace yourself, here comes something astounding:
+
+***We can nest nested data structures in other nested data structures***
+
+Your `Array` of `Hashes` (AoH) can have keys that point to `Array`s of
+`Array`s (AoA). Or your nested node structure of `Hash`es of `Hash`es (HoH)
+can have a key that points to another `Array` of `Hash`es (AoH). From what
+we now know about NDS, we should be able to see that these NDS' can become
+very complex!
+
+In this lesson, we're going to walk you through an improved vending
+machine model. Get a pen and some paper so that you can draw
+pictures to help drive these points home. Using sketches will really
+help you "get it." This is why developers in professional settings are
+always sketching their NDS' whiteboards. It really works!
+
+## Identify Nested Structures Can Be Mixed
+
+As we try to more accurately model the world around us, we'll see that it's
+very helpful to mix and match NDS to provide a better model of reality.
+
+Let's update our model of a grid-based vending machine again. Remember,
+they look like this:
+
+![Grid-based vending machine](https://curriculum-content.s3.amazonaws.com/programming-univbasics-5/nested-arrays-lab/vending_grid.png)
+
+We're going to work "top-down" and then re-work the modeling "bottom-up."
+Neither method is more correct than the other. The problem of modeling sometimes
+requires us to be flexible. "Top-down" means: describe the outermost structure
+and then fill it in, and fill in any structures that were just added until you
+reach the smallest structure. "Bottom-up" means: describe the innermost structure
+and "wrap it" with a container repeatedly until the outermost container is reached.
+
+### Array of Array of Array or Hashes...
+
+Ultimately, in order to model a real grid-based vending machine (C20 is a candy bar
+gum, B50 is a drink), we think the right NDS is:
+
+```text
+Array of Array
+...of Array
```

Comment:
why three arrays? Why not AoAoH? A grid with each cell being a hash of item data? I think the extra level is confusing.

Reply:
Because vending machines don't usually have only one item on the "spinner" (as we build on later). Also we're trying to move from simple case (earlier in the brick) AoA to something more complex..need another degree of complexity.